src/docfx/template/TemplateEngine.cs

```diff
@@ -60,6 +66,17 @@ public TemplateEngine(ErrorBuilder errors, Config config, PackageResolver packag
             _mustacheTemplate = new(_package, "ContentTemplate", _global);
         }
 
+        public TemplateEngine(ErrorBuilder errors, string locale, Package package)
```

Comment:
Can this be extracted to another class? Then no need to make all the missing fields nullable.